ui: center spinner components

xinhaoz opened the pull request:
This commit centers the spinner components used in db console. It also transforms the cluster summary card loading state into a skeleton instead of a spinner.

Epic: none

Release note: None

-------------------------------------

ui: fix skeleton loading heights

There were several places where are were using
the Skeleton antd component to show loading states.
Some of these were incorrectly using a paragraph
style placeholder when it should have been a single
line placeholder state. This commit fixes areas
where we don't want to use paragraph style placeholders.

Epic: none

Release note: None

xinhaoz committed:
ui: center spinner components

This commit centers the spinner components used in
db console. It also transforms the cluster summary
card loading state into a skeleton instead of a spinner.

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/133375)
<!-- Reviewable:end -->

xinhaoz commented:
https://www.loom.com/share/0cc1f0ab857b4c1eae8c7d9367db70a7

xinhaoz committed:
ui: fix skeleton loading heights

There were several places where are were using
the Skeleton antd component to show loading states.
Some of these were incorrectly using a paragraph
style placeholder when it should have been a single
line placeholder state. This commit fixes areas
where we don't want to use paragraph style placeholders.

Epic: none

Release note: None

dhartunian commented on pkg/ui/workspaces/cluster-ui/src/components/tooltipMessages/tableMetadataLastUpdatedTooltip.tsx:
```diff
@@ -49,12 +40,35 @@ const formatErrorMessage = (
 }: Props) => {
-  const durationText = timestamp?.fromNow() ?? "Never";
+  const duration = (
+    <span>
```
why is this wrapper necessary?

xinhaoz replied:
It's to wrap the skeleton component when it's in loading state so that it doesn't go onto a new line.

dhartunian commented on pkg/ui/workspaces/cluster-ui/src/databaseDetailsV2/tablesView.tsx:
```diff
@@ -123,7 +123,10 @@ const COLUMNS: (TableColumnProps<TableRow> & {
-      <RegionNodesLabel nodesByRegion={t.nodesByRegion} />
+      <RegionNodesLabel
+        loading={t.nodesByRegion.isLoading}
+        nodesByRegion={t.nodesByRegion.data}
```
why not pass `t.nodesByRegion` and let the component decide whether to do anything with `.isLoading`?

dhartunian commented on pkg/ui/workspaces/cluster-ui/src/databaseDetailsV2/types.ts:
```diff
@@ -8,6 +8,9 @@ import { NodeID } from "src/types/clusterTypes";
   qualifiedNameWithSchema: string;
-  nodesByRegion: Record<string, NodeID[]>;
+  nodesByRegion: {
+    isLoading: boolean;
```
There's no standard abstraction available around an "async loaded value"? Seems surprising to me to pepper the code with `isLoading` booleans.

xinhaoz replied:
There isn't one today, no. swr explicitly provides the loading/validating state separately from the resolved promise value. I'm also treating this as more of a special case since the nodes information currently loads separately from the rest of the table metadata -- we could think about putting this as part of the job so we don't have to go through the nodes api like we do today.

xinhaoz commented:
TFTR!
bors r+
